chapters/memory.txt

```diff
@@ -3294,6 +3458,86 @@ include::{generated}/api/protos/vkUnmapMemory.txt[]
+--
+Alternatively, to unmap a memory object once host access to it is no longer
+needed by the application, call:
+
+include::{generated}/api/protos/vkUnmapMemory2EXT.txt[]
+
+  * pname:device is the logical device that owns the memory.
+  * pname:pMemoryUnmapInfo is a pointer to a slink:VkMemoryUnmapInfoEXT
+    structure describing parameters of the unmap.
+
+This function behaves identically to flink:vkUnmapMemory except that it gets
+its parameters via an extensible structure pointer rather than directly as
+function arguments.
+
+include::{generated}/validity/protos/vkUnmapMemory2EXT.txt[]
+--
+
+[open,refpage='VkMemoryUnmapInfoEXT',desc='Structure containing parameters of a memory unmap operation',type='structs']
+--
+The sname:VkMemoryUnmapInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkMemoryUnmapInfoEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:memory is the slink:VkDeviceMemory object to be unmapped.
+
+.Valid Usage
+****
+  * pname:memory must: be currently host mapped
+ifdef::VK_EXT_map_memory_placed[]
+  * If ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT is set in pname:flags,
+    the <<features-memoryUnmapReserve, pname:memoryUnmapReserve>>
+    must: be enabled
+ifdef::VK_EXT_external_memory_host[]
+  * If ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT is set in pname:flags, the
+    memory object must: not have been imported from a handle type of
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT or
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT
+endif::VK_EXT_external_memory_host[]
+endif::VK_EXT_map_memory_placed[]
+****
+
+include::{generated}/validity/structs/VkMemoryUnmapInfoEXT.txt[]
+--
+
+ifdef::VK_EXT_map_memory_placed[]
+[open,refpage='VkMemoryUnmapFlagBitsEXT',desc='Bitmask specifying additional parameters of a memory unmap',type='enums']
+--
+Bits which can: be set in slink:VkMemoryUnmapInfoEXT::pname:flags, specifying
+additional properties of a memory unmap, are:
+
+include::{generated}/api/enums/VkMemoryUnmapFlagBitsEXT.txt[]
+
+  * ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT requests that virtual address
+    range currently occupied by the memory map be remain reserved after the
+    flink:vkUnmapMemory2EXT call completes.
+    Future system memory map operations or calls to flink:vkMapMemory or
+    flink:vkMapMemory2EXT will not return addresses in that range unless
+    the range has since been unreserved by the client or the memory map
+    operation is done with ename:VK_MEMORY_MAP_PLACED_BIT_EXT or the system
+    memory map equivalent.
+    When ename:VK_MEMORY_UNMAP_RESERVE_BIT_EXT is set, the memory unmap
+    operation may fail, in which case the memory object will remain host
+    mapped and flink:vkUnmapMemory2EXT will return
+    ename:VK_ERROR_MEMORY_MAP_FAILED.
```

Comment:
Is it possible to quantify (perhaps in a note) the circumstances under which this happens?

Reply:
Without digging into kernel source code, I'm not sure when it would fail.  I'm not even sure it can.  But `mmap()` is allowed to return `MAP_FAILED` for pretty arbitrary reasons and I would rather throw an error than silently fall back to `munmap()` and lose the reservation.

Reply:
Context: https://gitlab.freedesktop.org/jekstrand/mesa/-/commit/348a18c3a559e280d8ef859fc6079ec44c1bde2e#637da2a63a937bb34d4a7112d341f9ea5c652765_168_177

Reply:
Ok, if this is basically esoteric OS behavior then there's not much we can state here.